NitroxClient/Communication/Packets/Processors/CyclopsChangeColorProcessor.cs

```diff
@@ -0,0 +1,57 @@
+﻿using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxClient.GameLogic.Helper;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using System;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class CyclopsChangeColorProcessor : ClientPacketProcessor<CyclopsChangeColor>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsChangeColorProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsChangeColor colorPacket)
+        {
+            Optional<GameObject> opCyclops = GuidHelper.GetObjectFrom(colorPacket.Guid);
+
+            if (opCyclops.IsPresent())
+            {
+                CyclopsNameScreenProxy ScreenProxy = opCyclops.Get().GetComponentInChildren<CyclopsNameScreenProxy>();
+
+                if (ScreenProxy != null)
+                {
+                    SubName subname = (SubName)ScreenProxy.subNameInput.ReflectionGet("target");
+
+                    if (subname != null)
+                    {
+                        Vector3 hsb = new Vector3(colorPacket.HSB.X, colorPacket.HSB.Y, colorPacket.HSB.Z);
+                        Color color = new Color(colorPacket.Color.R, colorPacket.Color.G, colorPacket.Color.B, colorPacket.Color.A);
+                        subname.SetColor(colorPacket.Index, hsb, color);
+                        ScreenProxy.subNameInput.ReflectionCall("SetColor", false, false, new object[] { colorPacket.Index, color });
```

Comment:
Can you also sync the tab that is currently open (`BASE`, `STRIPE 1`, `STRIPE 2` and `NAME`)?

Should be as easy as calling `ScreenProxy.subNameInput.SetSelected(colorPacket.Index)` here, but then you need to send a color packet as soon as a tab is clicked. This also fixes the "currently selected" color containing the text for the wrong color.